usefulUtilities/sit/sitClient.js

```diff
@@ -229,78 +233,68 @@
-        if (standUpWrapperCalled) {
-            return;
-        }
-        standUpWrapperCalled = true;
-        Script.setTimeout(function(){
-            standUp();
-        }, STAND_UP_WRAPPER_WAIT_MS);
-    }
-
     // Standup functionality
-    var STANDUP_BUMP = 0.225;
-    var roles = [];
-    function standUp() {
-        
+    var alreadyCalledStandUp = false;
+    function standUp(id, args) {
         if (DEBUG) {
-            console.log("standup from ", _this.entityID);
-        }
-        if (standUpWrapperCalled) {
-            standUpWrapperCalled = false;
+            console.log("sitClient.js: standup from seat ID: " + _this.entityID + "\nCall has args: " + (args && args[0]));
         }
 
-        // get the entityID, previous position and orientation 
-        var settingsEntityID = Settings.getValue(SETTING_KEY_AVATAR_SITTING);
-        changedSeats = false;
-        // STANDING FROM THIS CHAIR
-        // Make avatar stand up (if changed seat do not do this)
-        if (settingsEntityID === _this.entityID) { // POSSIBLE RACE CONDITION WITH SETTINGS BEING CHANGED BY NEW SEAT
-            Settings.setValue(SETTING_KEY_AVATAR_SITTING, null);
-            standAvatarUp();
-        }
+        if (isSittingInThisChair) {
+            if (DEBUG) {
+                console.log("sitClient.js: we are indeed sitting in this chair.");
+            }
 
-        // RESET SETTINGS FOR THIS CHAIR
-        // Could have changed seats, keep avatar sitting if did not go through above procedure
-        if (!_this.locked) {
-            Entities.editEntity(_this.entityID, { locked: false });
-        }
+            if (alreadyCalledStandUp) {
+                if (DEBUG) {
+                    console.log("sitClient.js: we already called standup for: " + _this.entityID + "; no action taken");
+                }
+                return;
+            }
 
-        _this.driveKeyPressedStart = false;
-        _this.sitDownSettlePeriod = false;
+            alreadyCalledStandUp = true;
+            
+            stopUpdateInterval();
+            _this.sitDownSettlePeriod = false;
 
-        Entities.callEntityServerMethod(_this.entityID, "onStandUp");
+            if (isActionEventConnected) {
+                Controller.actionEvent.disconnect(onActionEvent);
+                isActionEventConnected = false;
+            }
+            MyAvatar.endSit(MyAvatar.position, MyAvatar.orientation);
+            
+            Settings.setValue(SETTING_KEY_AVATAR_SITTING, null);
+            isSittingInThisChair = false;
 
-        Entities.callEntityServerMethod(
-            _this.entityID,
-            "addAllOtherSittableOverlays",
-            AvatarList.getAvatarsInRange(_this.seatCenterPosition, CAN_SIT_M)
-        );
-        createSittableUI();
-        stopUpdateInterval();
+            // RESET SETTINGS FOR THIS CHAIR
+            // Could have changed seats, keep avatar sitting if did not go through above procedure
```

Comment:
I think you can remove this comment now

Reply:
ACK

Reply:
oh, this comment is almost correct, it's just slightly misleading. when the comment says "settings", it means "locked state". i'll update

Reply:
oh, unless you were only talking about the second line here, in which case yeah, i gotta remove that